test/xbgpu/test_engine.py

```diff
@@ -891,7 +923,12 @@ async def _send_data(
                 assert ig_recv["timestamp"].value == index * timestamp_step
                 assert ig_recv["frequency"].value == frequency
                 assert ig_recv["beam_ants"].value == np.sum(present[index])
-                beam_results[i, j, ...] = ig_recv["bf_raw"].value
+                beam_results[beam_output.name][j, ...] = ig_recv["bf_raw"].value
+
+            if beam_capture_stop_heap_index:
```

Comment:
This `if` can be removed - we might as well test this for all the unit tests, to ensure we've calculated batch_indices correctly.